gov2/iam/scenarios/scenario_assume_role.go

```diff
@@ -0,0 +1,314 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package scenarios
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/iam/types"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/aws/smithy-go"
+	"github.com/awsdocs/aws-doc-sdk-examples/gov2/demotools"
+	"github.com/awsdocs/aws-doc-sdk-examples/gov2/iam/actions"
+)
+
+// IScenarioHelper abstracts input and wait functions from a scenario so that they
+// can be mocked for unit testing.
+type IScenarioHelper interface {
+	GetName() string
+	Pause(secs int)
+}
+
+const rMax = 100000
+type ScenarioHelper struct {
+	Prefix string
+	Random *rand.Rand
+}
+
+// GetName returns a unique name formed of a prefix and a random number.
+func (helper *ScenarioHelper) GetName() string {
+	return fmt.Sprintf("%v%v", helper.Prefix, helper.Random.Intn(rMax))
+}
+
+// Pause waits for the specified number of seconds.
+func (helper ScenarioHelper) Pause(secs int) {
```

Comment:
Is `helper` needed here?

Reply:
Yes, because I pass a mock version for unit testing that implements Pause as a no-op.

Reply:
Ok. This is a gap in my GO knowledge :p